flocker/ca/test/test_ca.py

```diff
@@ -352,14 +352,24 @@ def test_certificate_subject_control_service(self):
         self.assertEqual(
             subject.CN, b"control-service")
 
-    def test_subjectAltName(self):
+    def test_subjectAltName_dns(self):
         """
         The generated certificate has a subjectAltName containing the given
-        hostname.
+        hostname as a DNS record.
         """
         assert_has_extension(self, self.credential.credential,
                              b"subjectAltName", b"DNS:control.example.com")
 
+    def test_subjectAltName_ipv4(self):
+        """
+        The generated certificate has a subjectAltName containing the given
```

Comment:
Reading the test docstring for this and `test_subjectAltName_ipv4`, they can't both be true. Put "if ..." at the end of each.